### PR TITLE
More documentation for SQL compiler; fixes for unicode strings and identifiers

### PR DESCRIPTION
**Pull request by mihaibudiu**

### CODE REVIEW

File: docs/sidebars.js [hunk 1]
```diff
@@ -44,14 +44,19 @@ const sidebars = {
...
       items: [
-        "sql/types",
-        "sql/boolean",
-        "sql/comparisons",
-        "sql/integer",
-        "sql/float",
-        "sql/decimal",
-        "sql/fp",
-        "sql/string",
+          "sql/grammar",
```
Comment (gz) [inline docs/sidebars.js:47]: was this formatted using `yarn format`?

File: docs/docs/sql/dbsp-workflow.svg
Comment (gz) [inline docs/docs/sql/dbsp-workflow.svg:1]: this looks very strange hopefully it's just github preview that's broken. we should make sure this works in dark mode and has a transparent background (doesn't look like it?)

Comment (gz) [reply docs/docs/sql/dbsp-workflow.svg:1]: We should also make sure that all diagrams we produce have a similar look&feel. I've used https://excalidraw.com/ in the past for this, it allows you to easily create such figures and they look good.

**Review by gz — APPROVED** [review]

File: docs/sidebars.js (hunk 1 above)
Comment (mihaibudiu) [reply docs/sidebars.js:47]: no.
who remembers what tools one needs to use?
it should be part of the commit process if it's necessary

File: docs/docs/sql/dbsp-workflow.svg
Comment (mihaibudiu) [reply docs/docs/sql/dbsp-workflow.svg:1]: this is done with powerpoint, the output looks ok for me

Comment (mihaibudiu) [reply docs/docs/sql/dbsp-workflow.svg:1]: transparent backgrounds usually don't work well with themes. look at the feldera logo on the top-left.
I may tweak the theme, I find that the contrast is too low on quoted fragments (```).

File: docs/sidebars.js (hunk 1 above)
Comment (gz) [reply docs/sidebars.js:47]: It's documented in the readme but I'll add it to CI

File: docs/docs/sql/dbsp-workflow.svg
Comment (gz) [reply docs/docs/sql/dbsp-workflow.svg:1]: The logo shouldn't be black but blue... And we should serve a slightly different colored logo (lighter shade of blue) for dark mode.

**Pull request merged**